qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/fitting_widgets/basic_fitting/basic_fitting_model.py

```diff
@@ -783,14 +814,27 @@ def _evaluate_plot_guess(self, plot_guess: bool) -> str:
 
     def _evaluate_function(self, fit_function: IFunction, output_workspace: str) -> str:
         """Evaluate the plot guess fit function and returns the name of the resulting guess workspace."""
+        if self.fitting_context.plot_guess_type == 'x from fit range':
+            data_ws = retrieve_ws(self.current_dataset_name)
+            data = np.linspace(self.current_start_x, self.current_end_x, data_ws.getNumberBins())
+        elif self.fitting_context.plot_guess_type == 'Uniform points across data range':
+            data_ws = retrieve_ws(self.current_dataset_name)
+            data = np.linspace(data_ws.dataX(0)[0], data_ws.dataX(0)[-1], self.fitting_context.plot_guess_points)
+        elif self.fitting_context.plot_guess_type == 'Custom x range':
+            data = np.linspace(self.fitting_context.plot_guess_start_x, self.fitting_context.plot_guess_end_x,
+                               self.fitting_context.plot_guess_points)
+        else:
+            raise ValueError('Plot guess type: ' + self.fitting_context.plot_guess_type + ' is not recognised')
```

Comment:
Suggestion to use an f-string
```suggestion
            raise ValueError(f"Plot guess type '{self.fitting_context.plot_guess_type}' is not recognised.")
```